File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.js
function solution(myString, pat) {
    let answer = 0;
    for (let i = 0; i <= myString.length - pat.length; i++) {
        if (myString.substring(i, i + pat.length) === pat) {
            answer++;
        }
    }
    return answer;
}

File: 프로그래머스/0/181871. 문자열이 몇 번 등장하는지 세기/문자열이 몇 번 등장하는지 세기.c
#include <stdio.h>
#include <string.h>

int solution(const char* myString, const char* pat) {
    int answer = 0;
    int lenMy = strlen(myString);
    int lenPat = strlen(pat);
    
    for (int i = 0; i <= lenMy - lenPat; i++) {
        if (strncmp(&myString[i], pat, lenPat) == 0) {
            answer++;
        }
    }
    
    return answer;
}
